--- a/src/hotspot/share/utilities/globalDefinitions_gcc.hpp	2019-04-02 20:25:37.000000000 -0500
+++ b/src/hotspot/share/utilities/globalDefinitions_gcc.hpp 	2019-04-22 18:35:47.809350313 -0500
@@ -218,8 +219,20 @@
 
 // Checking for finiteness
 
+#ifdef __APPLE__
+#include <TargetConditionals.h>
+#else
+#define TARGET_OS_IPHONE 0
+#endif
+
+#if TARGET_OS_IPHONE
+#define USE_LIBRARY_BASED_TLS_ONLY
+inline int g_isfinite(jfloat  f)                 { return isfinite(f); }
+inline int g_isfinite(jdouble f)                 { return isfinite(f); }
+#else
 inline int g_isfinite(jfloat  f)                 { return finite(f); }
 inline int g_isfinite(jdouble f)                 { return finite(f); }
+#endif
 
 
 // Wide characters
